### PR TITLE
Added Message#extractChannels() to get the Mentioned ServerTextChannels from a Message

### PR DESCRIPTION
**Pull request by DerTobi01**
Because I really wanna contribute x_x
lolz

### CODE REVIEW

File: src/main/java/de/btobastian/javacord/entities/message/Message.java [hunk 1]
```diff
@@ -984,6 +985,27 @@ default String getReadableContent() {
...
+     * Gets a collection of all <code>ServerTextChannels</code> mentioned in the message.
+     *
+     * @return The collection of
+     */
+    default List<ServerTextChannel> extractChannels() {
+        String content = getContent();
+        List<ServerTextChannel> channels = new ArrayList<>();
+        List<String> matches = new ArrayList<>();
+        Optional<Channel> chlOpt;
+        Matcher matcher = Pattern.compile("(<#)[0-9]+(>)")
```
Comment (xaanit) [inline src/main/java/de/btobastian/javacord/entities/message/Message.java]: Why not do `"<#([0-9]+)>"`

File: src/main/java/de/btobastian/javacord/entities/message/Message.java [hunk 2]
```diff
@@ -984,6 +985,27 @@ default String getReadableContent() {
...
+                .matcher(content);
+        while (matcher.find()) {
+            matches.add(matcher.group());
+        }
+        for (String x : matches) {
+            getApi().getChannelById(x.substring(x.indexOf("<#")+2, x.indexOf(">"))).ifPresent(a -> a.asServerTextChannel().ifPresent(channels::add));
```
Comment (xaanit) [inline src/main/java/de/btobastian/javacord/entities/message/Message.java]: Using the aforementioned pattern you could just use the matcher.group(), you could also just merge the while and for loop.

File: src/main/java/de/btobastian/javacord/entities/message/Message.java [hunk 3]
```diff
@@ -984,6 +985,25 @@ default String getReadableContent() {
...
+    default List<ServerTextChannel> extractChannels() {
+        String content = getContent();
+        List<ServerTextChannel> channels = new ArrayList<>();
+        String x;
+        Optional<Channel> chlOpt;
+        Matcher matcher = Pattern.compile("<#([0-9]+)>")
+                .matcher(content);
+        while (matcher.find()) {
+            x = matcher.group();
+            getApi().getChannelById(x.substring(x.indexOf("<#")+2, x.indexOf(">"))).ifPresent(a -> a.asServerTextChannel().ifPresent(channels::add));
```
Comment (xaanit) [inline src/main/java/de/btobastian/javacord/entities/message/Message.java]: You didn't change this at all lol, x will only be the numbers with my pattern.

Comment (DerTobi01) [reply src/main/java/de/btobastian/javacord/entities/message/Message.java]: it works like this 🤔 
grafik

Comment (xaanit) [reply src/main/java/de/btobastian/javacord/entities/message/Message.java]: Cause regex is weird.

Using just `group()` will grab the whole string (group(0)). Use `.group(1)` instead and you can eliminate `x` entirely by just calling the method in the `getChannelById`

**Comment by BtoBastian** [comment]
Hey,
thank you, but please take a look at Javacord3's [contribution guidelines](https://github.com/BtoBastian/Javacord3-Docs/blob/master/.github/CONTRIBUTING.md).
There a serveral issues with this PR:
* Instead of compiling your own regex pattern every time, you should use the precompiled pattern from [DiscordRegexPattern.java](https://github.com/BtoBastian/Javacord/blob/v_3/src/main/java/org/javacord/util/DiscordRegexPattern.java)
* The code doesn't follow Javacord's code conventions (e.g. calling a variable just `x`)
* The JavaDoc comment is just a half sentence (`@return The collection of`)
* All returned collections in Javacord should be immutable
* The method name doesn't really fit to the other used methods names (e.g. `#getMentionedUsers()`, `#getMentionedRoles()`, `#getCustomEmojis()`, ...)
* You can mention voice channels, even though they are not displayed in the auto completion of the client (not sure about categories).

**Comment by DerTobi01** [comment]
@BtoBastian when editing the stuff to what you needed, i realized this is already in there xd didnt notice.

/closed

**Comment by Vampire** [comment]
And how is it different from `Message#getMentionedChannels`? (Didn't look too close at the changes)

**Comment by BtoBastian** [comment]
I didn't even notice that this method already exists.
Just assumed it doesn't. ^^

**Comment by FunkFrog** [comment]
_why._

**Pull request closed without merging**